sklearn/preprocessing/_encoders.py

```diff
@@ -398,9 +401,25 @@ def transform(self, X):
             # categories that occur after them to avoid an empty column.
 
             keep_cells = X_int != to_drop
+
+            if isinstance(self.drop, str) and self.drop == 'if_binary':
+                keep_cells = keep_cells.T
+                X_int = X_int.T
+                for i in range(n_features):
+                    if len(self.categories_[i]) > 2:
+                        keep_cells[i] = True
```

Comment:
If keep_cells is not transposed, you should be able to use: `keep_cells[:, i] = True`